### PR TITLE
finished flashcard project

### PR DESCRIPTION
**Pull request by fgaspar17**

### CODE REVIEW

**Comment by stevietv** [comment]
Hi @fgaspar17 - thanks for your submission. Before I review this any further can you take a look at something? I defined my database with a connection string like this:

```
    "ConnectionStrings": {
        "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=flashcards-fgaspar;Trusted_Connection=True",
        "SetupConnection": "Server=(localdb)\\mssqllocaldb;Trusted_Connection=True"
    },
```

but in your DatabaseConfig.cs class you hardcode the database name to `flashcardsdb` - this means that my further connections fail. Is it a firm requirement to have this name for the db, or can you incorporate a variable into the appsettings.json to allow for flexible db name?

**Comment by fgaspar17** [comment]
Hey @stevietv, I added a variable to the config file, so you can name the database whatever you want. I hadn't thought of that when working on the project. Thank you for your review.

File: Flashcards.fgaspar17/FlashcardsLibrary/DatabaseConfig.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        try
+        {
+            using IDbConnection connection = new SqlConnection(GlobalConfig.SetupConnectionString);
+
+            connection.Open();
+            string? db = connection.QuerySingleOrDefault<string>(@$"SELECT name FROM sys.databases WHERE name = '{databaseName}'");
+
+            if (db == null)
+            {
+                connection.Execute($"CREATE DATABASE {databaseName};");
```
Comment (stevietv) [inline Flashcards.fgaspar17/FlashcardsLibrary/DatabaseConfig.cs:21]:
```suggestion
                connection.Execute($"CREATE DATABASE [{databaseName}];");
```

using the square brackets ensures that if a user enters a special character as their database name that an SQL exception isn't thrown

**Review by stevietv — APPROVED** [review]
Hey @fgaspar17 - Really nicely handled. Great project!
You've got a great structure of files and folders, its easy to follow. Everything runs smoothly too. 

Nice work on the challenge too! Looking forward to seeing more from you.

**Pull request closed without merging**